src/core/xconnection.rs

```diff
@@ -829,17 +838,25 @@ impl XConn for XcbConnection {
     }
 
     fn grab_keys(&self, key_bindings: &KeyBindings) {
+        let mut modifiers = vec![0];
+
+        if let Some(&numlock_mask) = self.modifier_map.get("Num_Lock") {
```

Comment:
Is there ever a case where we _wouldn't_ have a NumLock modifier available?

Reply:
I honestly don't know, maybe if someone would intentionally remove `Num_Lock` from `mod2` in some config file?